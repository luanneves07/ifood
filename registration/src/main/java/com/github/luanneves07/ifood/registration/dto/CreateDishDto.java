package com.github.luanneves07.ifood.registration.dto;

import java.math.BigDecimal;

public class CreateDishDto {

	public String name;

	public String description;

	public BigDecimal price;

}
