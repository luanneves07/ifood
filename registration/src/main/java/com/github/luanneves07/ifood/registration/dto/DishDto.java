package com.github.luanneves07.ifood.registration.dto;

import java.math.BigDecimal;

public class DishDto {

	public String name;

	public BigDecimal price;

	public String description;

	public RestaurantDto restaurant;
}
