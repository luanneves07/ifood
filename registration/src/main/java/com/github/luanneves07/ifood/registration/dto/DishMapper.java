package com.github.luanneves07.ifood.registration.dto;

import org.mapstruct.Mapper;
import org.mapstruct.factory.Mappers;

import com.github.luanneves07.ifood.registration.Dish;

@Mapper
public interface DishMapper {
	
	public static final DishMapper INSTANCE = Mappers.getMapper(DishMapper.class);

	public DishDto dishToDishDto(Dish dish);

	public Dish toDish(DishDto dishDto);
}
