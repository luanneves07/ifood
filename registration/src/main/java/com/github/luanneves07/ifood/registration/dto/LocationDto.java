package com.github.luanneves07.ifood.registration.dto;

public class LocationDto {
	
	public Double latitude;

	public Double longitude;
}
