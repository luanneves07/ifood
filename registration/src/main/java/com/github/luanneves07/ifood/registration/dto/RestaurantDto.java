package com.github.luanneves07.ifood.registration.dto;

public class RestaurantDto {

	public String owner;

	public String cnpj;

	public String tradingName;

	public String creationTime;

	public LocationDto location;

	@Override
	public String toString() {
		return String.format(":::%s:::%n::<tradingName>%s::%n::<owner>%s::%n::<creationTime>%s::%n",
				this.getClass().getSimpleName(), tradingName, owner, creationTime);
	}
}
