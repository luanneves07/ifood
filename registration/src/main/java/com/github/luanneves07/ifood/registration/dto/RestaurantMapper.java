package com.github.luanneves07.ifood.registration.dto;

import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.factory.Mappers;

import com.github.luanneves07.ifood.registration.Restaurant;

@Mapper
public interface RestaurantMapper {

	public static final RestaurantMapper INSTANCE = Mappers.getMapper(RestaurantMapper.class);

	@Mapping(source = "name", target = "tradingName")
	public RestaurantDto restaurantToRestaurantDto(Restaurant restaurant);

	@Mapping(source = "tradingName", target = "name")
	public Restaurant toRestaurant(RestaurantDto restaurant);

}
