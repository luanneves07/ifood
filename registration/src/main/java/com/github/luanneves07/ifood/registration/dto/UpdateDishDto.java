package com.github.luanneves07.ifood.registration.dto;

import java.math.BigDecimal;

public class UpdateDishDto {

	public BigDecimal price;

}
