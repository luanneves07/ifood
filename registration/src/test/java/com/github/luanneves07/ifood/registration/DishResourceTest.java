package com.github.luanneves07.ifood.registration;

import static org.junit.Assert.assertEquals;

import java.math.BigDecimal;
import java.util.Optional;

import javax.ws.rs.core.Response.Status;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import com.github.luanneves07.ifood.registration.dto.DishDto;

import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import io.restassured.RestAssured;
import io.restassured.http.ContentType;
import io.restassured.specification.RequestSpecification;

@DBRider
@QuarkusTest
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTestResource(RegistrationTestLifeCycleManager.class)
public class DishResourceTest {
	
	private RequestSpecification given() {
		return RestAssured.given().contentType(ContentType.JSON);
	}

	@Test
	@DataSet("dish-01.yml")
	public void testListDishes() {
		Long idRestaurant = 2L;
		String res = given()
				.with().pathParam("idRestaurant", idRestaurant)
				.when().get("/restaurants/{idRestaurant}/dishes")
				.then()
				.statusCode(Status.OK.getStatusCode())
				.extract().asString();
		Approvals.verifyJson(res);
	}

	@Test
	@DataSet("dish-01.yml")
	public void testCreateDish() {
		Long idRestaurant = 2L;
		DishDto dto = new DishDto();
		dto.name = "potato ahh";
		dto.price = new BigDecimal(150);
		
		String res = given()
			.with().pathParam("idRestaurant", idRestaurant)
			.body(dto)
			.when().post("/restaurants/{idRestaurant}/dishes")
			.then()
			.statusCode(Status.OK.getStatusCode())
			.extract().asString();
		
		Dish found = Dish.find("name", dto.name).firstResult();
		assertEquals(dto.name, found.name);
		Approvals.verifyJson(res);
	}

	@Test
	@DataSet("dish-01.yml")
	public void testUpdateRestaurants() {
		Long id = 456L;
		Long idRestaurant = 2L;
		DishDto dto = new DishDto();
		dto.name = "test01";
		dto.description = "Luan";
		dto.price = new BigDecimal(150);
		
		String res = given()
			.with().pathParams("idRestaurant", idRestaurant, "id", id)
			.body(dto)
			.when().put("/restaurants/{idRestaurant}/dishes/{id}")
			.then()
			.statusCode(Status.OK.getStatusCode())
			.extract().asString();
		
		Dish found = Dish.findById(id);
		assertEquals(dto.name, found.name);
		Approvals.verifyJson(res);
	}
	
	@Test
	@DataSet("dish-01.yml")
	public void testDeleteRestaurants() {
		Long id = 123L;
		Long idRestaurant = 2L;
		given()
			.with().pathParams("idRestaurant", idRestaurant, "id", id)
			.when().delete("/restaurants/{idRestaurant}/dishes/{id}")
			.then()
			.statusCode(Status.OK.getStatusCode())
			.extract().asString();
		
		Optional<Dish> found = Dish.findByIdOptional(id);
		assertEquals(true, found.isEmpty());
	}
}