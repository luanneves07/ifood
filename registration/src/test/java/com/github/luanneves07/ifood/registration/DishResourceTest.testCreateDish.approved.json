{
  "name": "potato ahh",
  "price": 150
}